tests/glsl/shader-subgroup-arithmetic.slang

```diff
@@ -0,0 +1,296 @@
+// not testing glsl or spirv due to crashing issue from some odd code emitting which looks right but fails
+// TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl
```

Comment:
This source code is in GLSL syntax.
It doesn't make sense that you cannot use GLSL as target for this test.
GLSL -> GLSL should work.

Reply:
the emitted code is read by slang incorrectly I think: attached is a picture of normal glsl which would compile on any normal glsl compiler, but on slang it does not
![image](https://github.com/jkwak-work/slang/assets/159081215/32d104e3-a35d-4c76-9f0f-3880899f39dd)